pulsar-client/src/main/java/org/apache/pulsar/client/impl/auth/oauth2/AuthenticationFactoryOAuth2.java

```diff
@@ -43,23 +44,53 @@ public static Authentication clientCredentials(URL issuerUrl, URL credentialsUrl
      *
-     * @param issuerUrl the issuer URL
+     * @param issuerUrl      the issuer URL
      * @param credentialsUrl the credentials URL
-     * @param audience An optional field. The audience identifier used by some Identity Providers, like Auth0.
-     * @param scope An optional field. The value of the scope parameter is expressed as a list of space-delimited,
-     *              case-sensitive strings. The strings are defined by the authorization server.
-     *              If the value contains multiple space-delimited strings, their order does not matter,
-     *              and each string adds an additional access range to the requested scope.
-     *              From here: https://datatracker.ietf.org/doc/html/rfc6749#section-4.4.2
+     * @param audience       An optional field. The audience identifier used by some Identity Providers, like Auth0.
+     * @param scope          An optional field. The value of the scope parameter is expressed as a list of
+     *                       space-delimited,
+     *                       case-sensitive strings. The strings are defined by the authorization server.
+     *                       If the value contains multiple space-delimited strings, their order does not matter,
+     *                       and each string adds an additional access range to the requested scope.
+     *                       From here: https://datatracker.ietf.org/doc/html/rfc6749#section-4.4.2
      * @return an Authentication object
      */
     public static Authentication clientCredentials(URL issuerUrl, URL credentialsUrl, String audience, String scope) {
+        return clientCredentials(issuerUrl, credentialsUrl, audience, scope, null, null, null);
+    }
+
+    /**
+     * Authenticate with client credentials.
+     *
+     * @param issuerUrl          the issuer URL
+     * @param credentialsUrl     the credentials URL
+     * @param audience           An optional field. The audience identifier used by some Identity Providers, like Auth0.
+     * @param scope              An optional field. The value of the scope parameter is expressed as a list of
+     *                           space-delimited,
+     *                           case-sensitive strings. The strings are defined by the authorization server.
+     *                           If the value contains multiple space-delimited strings, their order does not matter,
+     *                           and each string adds an additional access range to the requested scope.
+     *                           From here: https://datatracker.ietf.org/doc/html/rfc6749#section-4.4.2
+     * @param connectTimeout     the connect timeout in milliseconds
+     * @param readTimeout        the read timeout in milliseconds
+     * @param trustCertsFilePath the path to the file that contains trusted certificates
+     * @return an Authentication object
+     */
+    @Builder(builderMethodName = "clientCredentialsBuilder")
```

Comment:
It would be preferable not to use Lombok's Builder for Pulsar's client API since it's harder to maintain. 

We can use Lombok's Builder if we set the builderClassName to something meaningful.

By default, the builderClassName would be (ReturnTypeName)Builder, so that would be AuthenticationBuilder unless the return type is changed or builderClassName is set explicitly. The builder classname wouldn't contain the relevant context and maintaining the API backwards compatibility later would be harder if the builder class name isn't set to something useful.
In this case, since the builder is for AuthenticationOAuth2 with client credentials flow, one possibility would be to set `builderClassName` to `OAuth2WithClientCredentialsBuilder` (to keep it shorter).